### DISCUSSION TITLE
For Ch.4, how should we handle withdrawing bitcoin in the game?

### DISCUSSION THREAD

**Discussion by ecurrencyhodler** [saving-satoshi/saving-satoshi#368] [General]
In Ch. 4, players reap the rewards of their labor and are prompted to withdraw the bitcoin they mined from the mining pool.

To do this, they generate pubkey pair. Then they generate an address. Then they withdraw the bitcoin.

One suggestion was to use Signet. How difficult would this be to integrate into our game? 

What are some Signet alternatives?

**Comment by afterburn** [comment]
We'd need a way to broadcast transactions to either `signet` or `testnet` if we're set on using an actual network to transfer funds. There are API's that provide this functionality, but I'd like us to steer away from being too reliant on third-parties for our app to function properly. It'd probably mean running our own node, which could expose some endpoints that we can use to broadcasts txs through, though I'm curious to learn what @adamjonas thinks of being reliant on third-parties, perhaps its a non-issue.

The alternative is to fake everything (like we're already doing for mining), this is also simpler to implement but of course this approach takes away from the rewarding feeling somewhat. This could potentially be mitigated if we allow users to spend their fakecoins on the saving satoshi platform itself (gamification).

**Reply by ecurrencyhodler** [reply]
It seems like a lot of work and maintenance to self-host. I imagine it might be better to leverage API's. Only thing we would need to really keep an eye on is replenishing the signet coins that we are distributing. 

I also like the idea of someone checking the withdrawal on a block explorer somewhere too. Makes it feel more real.

**Reply by ecurrencyhodler** [reply]
I found a faucet that has an api: https://github.com/jsarenik/bitcoin-faucet-shell
Also this explorer supports signet: https://mempool.space/signet

**Comment by ecurrencyhodler** [comment]
We will not be using signet for the current version of the game. We will gather feedback to see if players prefer to use it once we go live.
